File: frontend/src/api/dto/teamPageSummaryDto.ts
import type { Visibility } from './visibility'

/**
 * Team page summary for listings
 */
export interface TeamPageSummaryDto {
  /** Page ID (TSID) */
  id: string
  /** Page title */
  title: string
  /** Page URL slug */
  slug: string
  /** Visibility level */
  visibility: Visibility
  /** Page order */
  order: number
  /** Whether the page is soft-deleted */
  deleted: boolean
}

File: backend/src/main/java/fr/pedalons/dto/pages/response/TeamPageSummaryDto.java
package fr.pedalons.dto.pages.response;

import fr.pedalons.common.TsidUtils;
import fr.pedalons.domain.team.TeamPage;
import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.Visibility;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Team page summary for listings")
@ValidateSchema
public record TeamPageSummaryDto(
    @Schema(description = "Page ID (TSID)", required = true) String id,
    @Schema(description = "Page title", required = true) String title,
    @Schema(description = "Page URL slug", required = true) String slug,
    @Schema(description = "Visibility level", required = true) Visibility visibility,
    @Schema(description = "Page order", required = true) int order,
    @Schema(description = "Whether the page is soft-deleted", required = true) boolean deleted) {

  public static TeamPageSummaryDto from(TeamPage page) {
    return new TeamPageSummaryDto(
        TsidUtils.toString(page.getId()),
        page.getName(),
        page.getSlug(),
        page.getVisibility(),
        page.getPageOrder() != null ? page.getPageOrder() : 0,
        page.isDeleted());
  }
}
